src/target/linux.rs

```diff
@@ -90,41 +94,88 @@ impl UnixTarget for LinuxTarget {
         self.pid
     }
 
-    fn unpause(&self) -> Result<nix::sys::wait::WaitStatus, Box<dyn std::error::Error>> {
+    fn unpause(&self) -> Result<WaitStatus, Box<dyn std::error::Error>> {
+        self.handle_breakpoint()?;
         ptrace::cont(self.pid(), None)?;
-        let status = nix::sys::wait::waitpid(self.pid(), None)?;
-        match status {
-            // We may have hit a user defined breakpoint
-            nix::sys::wait::WaitStatus::Stopped(_, nix::sys::signal::Signal::SIGTRAP) => {
-                if let Some(bp) = self
-                    .breakpoints
-                    .borrow_mut()
-                    .get_mut(&(self.read_regs()?.rip as usize - 1))
+        let status = waitpid(self.pid(), None)?;
+
+        //// We may have hit a user defined breakpoint
+        if let WaitStatus::Stopped(_, nix::sys::signal::Signal::SIGTRAP) = status {
+            if let Some(bp) = self
+                .breakpoints
+                .borrow_mut()
+                .get_mut(&(self.read_regs()?.rip as usize - 1))
+            {
+                // scope the borrow of hit_breakpoint
```

Comment:
This should not be necessary. `self.hit_breakpoint().borrow_mut()` will be dropped at the end of the statement if it isn't assigned to a variable.